scripts/components/dependencies_validator.ts

```diff
@@ -23,17 +23,27 @@ type DependencyViolation = {
 };
 
 /**
- * Validates dependencies. Inspects each path, enumerates all dependencies
+ * Validates dependencies.
+ *
+ * Inspects each path, enumerates all dependencies
  * using 'npm ls --all --json', i.e. direct and transitive, and validates them
  * against the rules.
+ *
+ * Inspects dependency version declarations in package json files
+ * in order to assert consistency.
  */
 export class DependenciesValidator {
   /**
    * Creates dependency validator
+   * @param packagePaths paths of packages to validate
+   * @param dependencyRules dependency exclusion and inclusion rules
+   * @param linkedDependencies dependencies that should be versioned with the same version
+   * @param execa in order to inject execa mock in tests
    */
   constructor(
     private packagePaths: Array<string>,
     private dependencyRules: Record<string, DependencyRule>,
+    private linkedDependencies: Array<Array<string>>,
```

Comment:
maybe name this `consistentVersionDependencies`? linkedDependencies makes me think there is some sort of functional connection between the dependencies.

Reply:
I borrowed terminology from https://github.com/changesets/changesets/blob/main/docs/linked-packages.md .
But maybe I understood it wrong and it should be "fixed" https://github.com/changesets/changesets/blob/main/docs/fixed-packages.md .

How about `sameVersionDependencies` ?

Reply:
Based on those docs, yes this should be linkedPackages. I'm fine with that.